src/clients/storage.ts

```diff
@@ -0,0 +1,64 @@
+import { Client, IAgentRuntime, ClientInstance } from "@elizaos/core";
+import * as Storage from '@web3-storage/w3up-client';
+import { StoreMemory } from '@web3-storage/w3up-client/stores/memory';
+import { Signer } from '@ucanto/principal/ed25519';
+import { StorageClientConfig, validateStorageClientConfig } from "../environments";
+import { defaultGatewayUrl, parseDelegation } from '../utils';
+
+export class StorageClient implements Client {
+    name = "storage";
+    private storageClient: Storage.Client | null = null;
+    config: StorageClientConfig | null = null;
+
+    async start(runtime: IAgentRuntime): Promise<ClientInstance> {
+        this.config = await validateStorageClientConfig(runtime);
+        this.storageClient = await createStorageClient(this.config);
+
+        return {
+            stop: async () => {
+                this.storageClient = null;
+                this.config = null;
+            }
+        };
```

Comment:
Shouldn't the client be returned here? If start is called again it'll overwrite the current one.

e.g. Farcaster does this: https://github.com/elizaos-plugins/client-farcaster/blob/4fc915babcf2fd1c4a826e306631d3aba617b6d6/src/farcaster-client.ts#L66-L81

Reply:
That is weird, because the `ClientInstance` type is not the actual instance, e.g:
```javascript
type ClientInstance = {
    /** Client name */
    /** Stop client connection */
    stop: (runtime: IAgentRuntime) => Promise<unknown>;
};
```

Reply:
@alanshaw I added a check to prevent the same client to be initialized again.

Reply:
I just think we should do what everyone else is doing...

https://github.com/elizaos-plugins/client-alexa/blob/3ebab5e228f0b9e0706419c2bf018a5404e796d4/src/client.ts#L14
https://github.com/elizaos-plugins/client-deva/blob/a28854d9ba4debf67edbc5a66b32aef19bf9b04e/src/client.ts#L22
https://github.com/elizaos-plugins/client-telegram-account/blob/ac425b95e27efa15c936e1cb627d9f0722a2bfa5/src/client.ts#L16

Reply:
sounds good @alanshaw ! I just updated the structure to match that pattern.